scaletest/templates/scaletest-runner/scripts/run.sh

```diff
@@ -54,13 +60,38 @@ for scenario in "${SCALETEST_PARAM_LOAD_SCENARIOS[@]}"; do
 			--job-timeout "${SCALETEST_PARAM_LOAD_SCENARIO_DASHBOARD_TRAFFIC_DURATION}m30s" \
 			--output json:"${SCALETEST_RESULTS_DIR}/traffic-dashboard.json" \
 			>"${SCALETEST_RESULTS_DIR}/traffic-dashboard-output.log"
+		status=$?
 		show_json "${SCALETEST_RESULTS_DIR}/traffic-dashboard.json"
 		;;
+	# Debug scenarios, for testing the runner.
+	"debug:success")
+		maybedryrun "$DRY_RUN" sleep 10
+		status=0
+		;;
+	"debug:error")
+		maybedryrun "$DRY_RUN" sleep 10
+		status=1
+		;;
 	esac
-	end_phase
+	set -e
+	if ((status > 0)); then
+		log "Load scenario failed: ${scenario} (exit=${status})"
+		failed+=(["${scenario}"]="$status")
+		PHASE_ADD_TAGS=error end_phase
+	else
+		end_phase
+	fi
 
 	wait_baseline "${SCALETEST_PARAM_LOAD_SCENARIO_BASELINE_DURATION}"
 done
 
+if ((${#failed} > 0)); then
+	log "Load scenarios failed: ${!failed[*]}"
+	for scenario in "${!failed[@]}"; do
```

Comment:
I didn't know you could do this :o

Reply:
Yup, it's the syntax for accessing associative array keys. 💪🏻 (I much prefer the Zsh way though `for scenario in ${(k)failed}` or even `for scenario code in ${(kv)failed}`. 😜)